core/src/main/java/com/onelogin/saml2/util/Util.java

```diff
@@ -550,191 +551,40 @@ public static String convertToPem(X509Certificate certificate) {
-
-		if (!parameters.isEmpty()) {
-			Boolean first = true;
-			for (Map.Entry<String, String> parameter : parameters.entrySet())
-			{
-				if (first) {
-					target += "?";
-					first = false;
-				} else {
-					target += "&";
-				}
-				target += parameter.getKey();
-				if (!parameter.getValue().isEmpty()) {
-					target += "=" + Util.urlEncoder(parameter.getValue());
-				}
-			}
-		}
-		response.sendRedirect(target);
-	}
-
-	/**
-	 * Redirect to location url
-	 * 
-	 * @param response
-	 * 				HttpServletResponse object to be used
-	 * @param location
-	 * 				target location url
-	 *
-	 * @throws IOException
-	 *
-	 * @see javax.servlet.http.HttpServletResponse#sendRedirect(String)
-	 */
-	public static void sendRedirect(HttpServletResponse response, String location) throws IOException {
-		Map<String, String> parameters  =new HashMap<String, String>();
-		sendRedirect(response, location, parameters);
-	}
-	
-	
-	/**
-	 * Returns the protocol + the current host + the port (if different than
-	 * common ports).
-	 *
-	 * @param request 
-	 * 				HttpServletRequest object to be processed
-	 *
-	 * @return the HOST URL
-	 */
-	public static String getSelfURLhost(HttpServletRequest request) {
-		String hostUrl = StringUtils.EMPTY;
-		final int serverPort = request.getServerPort();
-		if ((serverPort == 80) || (serverPort == 443) || serverPort == 0) {
-			hostUrl = String.format("%s://%s", request.getScheme(), request.getServerName());
-		} else {
-			hostUrl = String.format("%s://%s:%s", request.getScheme(), request.getServerName(), serverPort);
-		}
-		return hostUrl;
 	}
 
 	/**
-	 * @param request 
-	 * 				HttpServletRequest object to be processed
-	 *
-	 * @return the server name
-	 */
-	public static String getSelfHost(HttpServletRequest request) {
-		return request.getServerName();
-	}
-
-	/**
-	 * Check if under https or http protocol
+	 * Loads a resource located at a relative path
 	 *
-	 * @param request
-	 * 				HttpServletRequest object to be processed
+	 * @param relativeResourcePath
+	 *				Relative path of the resource
 	 *
-	 * @return false if https is not active
-	 */
-	public static boolean isHTTPS(HttpServletRequest request) {
-		return request.isSecure();
-	}
-
-	/**
-	 * Returns the URL of the current context + current view + query
-	 * 
-	 * @param request
-	 * 				HttpServletRequest object to be processed
+	 * @return the loaded resource in String format
 	 *
-	 * @return current context + current view + query
+	 * @throws IOException
 	 */
-	public static String getSelfURL(HttpServletRequest request) {
-		String url = Util.getSelfURLhost(request);
-
-		String requestUri = request.getRequestURI();
-		String queryString = request.getQueryString();
-
-		if (null != requestUri && !requestUri.isEmpty()) {		
-			url += requestUri;
+	public static String getFileAsString(String relativeResourcePath) throws IOException {
+		InputStream is = Util.class.getResourceAsStream("/" + relativeResourcePath);
```

Comment:
Updated this to work when loading classpath resources that are inside of JAR files (can't use `Files.readAllBytes` in that case). This is necessary because we want to load test resources of `java-saml-core` when running tests for `java-saml`.